timestamp-impl/src/main/java/com/palantir/timestamp/PersistentUpperLimit.java

```diff
@@ -51,13 +51,14 @@ public PersistentUpperLimit(TimestampBoundStore boundStore) {
      * @return upper limit timestamp
      */
     public long get() {
+        allocationFailures.verifyWeShouldIssueMoreTimestamps();
         return cachedValue;
     }
 
-    public synchronized long increaseToAtLeast(long minimum) {
+    public synchronized long increaseToAtLeast(long minimum, long buffer) {
```

Comment:
Could we add a bit of javadoc on this? It's not very clear without figuring out what we're trying to do when I should call this with `(foo+1M, 0)` and when I should call with `(foo, 1M)`